pkg/operator/controller/crl/crl_configmap.go

```diff
@@ -1,312 +0,0 @@
-		return false, nil, ctx, nil
-	}
-
-	if crls == nil {
-		crls = make(map[string]*pkix.CertificateList)
-	}
-
-	var subjectKeyIds []string
-	var nextCRLUpdate time.Time
-	now := time.Now()
-	for len(clientCAData) > 0 {
-		block, data := pem.Decode(clientCAData)
-		if block == nil {
-			break
-		}
-		clientCAData = data
-		cert, err := x509.ParseCertificate(block.Bytes)
-		if err != nil {
-			return false, nil, ctx, fmt.Errorf("client CA configmap has an invalid certificate: %w", err)
-		}
-		subjectKeyId := hex.EncodeToString(cert.SubjectKeyId)
-		if len(cert.CRLDistributionPoints) == 0 {
-			continue
-		}
-		if crl, ok := crls[subjectKeyId]; ok {
-			if crl.HasExpired(now) {
-				log.Info("certificate revocation list has expired", "subject key identifier", subjectKeyId)
-			} else {
-				subjectKeyIds = append(subjectKeyIds, subjectKeyId)
-				if (nextCRLUpdate.IsZero() || crl.TBSCertList.NextUpdate.Before(nextCRLUpdate)) && crl.TBSCertList.NextUpdate.After(now) {
-					nextCRLUpdate = crl.TBSCertList.NextUpdate
-				}
-				continue
-			}
-		}
-		log.Info("retrieving certificate revocation list", "subject key identifier", subjectKeyId)
-		if crl, err := getCRL(cert.CRLDistributionPoints); err != nil {
-			// Creating or updating the configmap with incomplete
-			// data would compromise security by potentially
-			// permitting revoked certificates.
-			return false, nil, ctx, fmt.Errorf("failed to get certificate revocation list for certificate key %s: %w", subjectKeyId, err)
-		} else {
-			crls[subjectKeyId] = crl
-			subjectKeyIds = append(subjectKeyIds, subjectKeyId)
-			log.Info("new certificate revocation list", "subject key identifier", subjectKeyId, "next update", crl.TBSCertList.NextUpdate.String())
-			if (nextCRLUpdate.IsZero() || crl.TBSCertList.NextUpdate.Before(nextCRLUpdate)) && crl.TBSCertList.NextUpdate.After(now) {
-				nextCRLUpdate = crl.TBSCertList.NextUpdate
-			}
-		}
-	}
-
-	if len(subjectKeyIds) == 0 {
-		return false, nil, ctx, nil
-	}
-
-	buf := &bytes.Buffer{}
-	for _, subjectKeyId := range subjectKeyIds {
-		asn1Data, err := asn1.Marshal(*crls[subjectKeyId])
-		if err != nil {
-			return false, nil, ctx, fmt.Errorf("failed to encode ASN.1 for CRL for certificate key %s: %w", subjectKeyId, err)
-		}
-		block := &pem.Block{
-			Type:  "X509 CRL",
-			Bytes: asn1Data,
-		}
-		if err := pem.Encode(buf, block); err != nil {
-			return false, nil, ctx, fmt.Errorf("failed to encode PEM for CRL for certificate key %s: %w", subjectKeyId, err)
-		}
-	}
-	crlData := buf.String()
-
-	crlConfigmapName := controller.CRLConfigMapName(ic)
-	crlConfigmap := corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      crlConfigmapName.Name,
-			Namespace: crlConfigmapName.Namespace,
-		},
-		Data: map[string]string{
-			"crl.pem": crlData,
-		},
-	}
-	crlConfigmap.SetOwnerReferences([]metav1.OwnerReference{ownerRef})
-
-	return true, &crlConfigmap, context.WithValue(ctx, "nextCRLUpdate", nextCRLUpdate), nil
-}
```

Comment:
I think you should keep the controller for now so that it can delete any stale configmap.
```suggestion
func desiredCRLConfigMap(ctx context.Context, ic *operatorv1.IngressController, ownerRef metav1.OwnerReference, clientCAData []byte, crls map[string]*pkix.CertificateList) (bool, *corev1.ConfigMap, context.Context, error) {
	// The ingress operator no longer needs to create a CRL configmap;
	// the router now downloads CRLs itself (see OCPBUGS-whatever).
	return false, nil, ctx, nil
}
```

Reply:
We should also have a to-do comment to remove the controller after 4.13 though.

Reply:
Good point. I will make that change. Do we need to consider EUS-to-EUS upgrades? If so, it may need to stay there to delete stale configmaps until after 4.14